### ISSUE TITLE
[FEATURE] mfsbdev as standard (TCP/Unix Socket) NBD server

### ISSUE THREAD

**Issue by tianon** [moosefs/moosefs#570]
Related to https://github.com/moosefs/moosefs/discussions/527#discussioncomment-5169723 and https://github.com/moosefs/moosefs/issues/113, I finally looked deeper at `mfsbdev` again, especially as I found https://manpages.debian.org/bookworm/qemu-system-x86/qemu-system-x86_64.1.en.html#NBD which makes it ~easy for QEMU to connect directly to an NBD server over TCP or Unix Domain Sockets, which seemed perfect!

However, I quickly re-realized that `mfsbdev` seems to be more of a helper for directly creating block devices on a host that are backed by MooseFS (via NBD), and I could not find a way to expose a "standard" NBD server from it.

So, in short, my feature request is that `mfsbdev` have some way to just expose a "proper" NBD server over TCP and/or unix socket instead of just the (AFAICT) custom protocol it currently supports. :bow: :heart:

**Comment by tianon** [comment]
A direct iSCSI server would also be interesting, especially as it could then be more easily used for things like booting a machine off the network with storage on MooseFS, but I think the iSCSI protocol is a lot more complicated than NBD (so that'd be a whole separate feature request / larger engineering effort, I think). :smile:

**Comment by ptempier** [comment]
I d guess the last over the top to implement is spdk nvme over fc.

Thats  what the guys at openebs did, no idea of the code behind it howerver,
cause their implementation doesnt require to use an actual nvme.

That's the high speed nvidia storage toolkit (spdk), 
using what seems to be the best block protocol ()nvme over fc at this moment.
